docs/api-guide/schemas.md

```diff
@@ -215,6 +215,110 @@ This also applies to extra actions for `ViewSet`s:
 project you may adjust `settings.DEFAULT_SCHEMA_CLASS`  appropriately.
 
+
+### Grouping Operations With Tags
+
+Tags can be used to group logical operations. Each tag name in the list MUST be unique. 
+
+---
+**Django REST Framework generates tags automatically with following logic:**
+1. Extract tag from `ViewSet`. 
+    1. If `ViewSet` name ends with `ViewSet`, or `View`; remove it.
+    2. Convert class name into lowercase words & join each word using a space. 
+    
+    Examples:
+    
+        ViewSet Class   |   Tags
+        ----------------|------------
+        User            |   ['user']	 
+        UserView        |   ['user']	 
+        UserViewSet     |   ['user']	
+        PascalCaseXYZ   |   ['pascal case xyz']
+        IPAddressView   |   ['ip address']
+
+2. If View is not an instance of ViewSet, tag name will be first element from the path. Also, any `-` or `_` in path name will be replaced by a space.
+Consider below examples.
+
+    Example 1: Consider a user management system. The following table will illustrate the tag generation logic.
+    Here first element from the paths is: `users`. Hence tag wil be `users`
+
+    Http Method                          |        Path       |     Tags
+    -------------------------------------|-------------------|-------------
+    PUT, PATCH, GET(Retrieve), DELETE    |     /users/{id}/  |   ['users']
+    POST, GET(List)                      |     /users/       |   ['users']
+    
+    Example 2: Consider a restaurant management system. The System has restaurants. Each restaurant has branches.
+    Consider REST APIs to deal with a branch of a particular restaurant.
+    Here first element from the paths is: `restaurants`. Hence tag wil be `restaurants`.
+    
+    Http Method                          |                         Path                       |     Tags
+    -------------------------------------|----------------------------------------------------|-------------------
+    PUT, PATCH, GET(Retrieve), DELETE:   | /restaurants/{restaurant_id}/branches/{branch_id}  |   ['restaurants']
+    POST, GET(List):                     | /restaurants/{restaurant_id}/branches/             |   ['restaurants']
+    
+    Example 3: Consider Order items for an e commerce company.
+    
+    Http Method                          |          Path           |     Tags
+    -------------------------------------|-------------------------|-------------
+    PUT, PATCH, GET(Retrieve), DELETE    |     /order-items/{id}/  |   ['order items']
+    POST, GET(List)                      |     /order-items/       |   ['order items']
+   
+
+---
+**You can override auto-generated tags by passing `tags` argument to the constructor of `AutoSchema`.**
+
+**`tags` argument can be a**
+1. list of string.
+    ```python
+   class MyView(APIView):
+        ...
+        schema = AutoSchema(tags=['tag1', 'tag2'])
+   ```
+2.  list of dict. This adds metadata to a single tag. Each dict can have 3 possible keys:
```

Comment:
How about we *start off* with only supporting the simple case? (Ie. list-of-strings)

Reply:
(It doesn't make a lot of sense to me to include the tag descriptions here, since you could perfectly well end up with two different views, both including the same tag, but including different descriptions for the same tag.)

Reply:
I got your point that tag description does not make sense here. I will remove this from here and allow the user to provide a custom schema generator class(a subclass of `openapi.SchemaGenerator`). In that class, dict tags would make more sense.

Reply:
That's sensible yup. Probably best to keep that out of this PR for now and address it seperatly, so that we can keep things minimal & niely reviewable.